aten/src/ATen/core/function_schema.h

```diff
@@ -373,6 +451,61 @@ struct FunctionSchema {
 
     return is_mutable(*index);
   }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs) const {
+    TORCH_INTERNAL_ASSERT(
+        (lhs.index < getCorrectList(lhs.type).size()),
+        "Invalid index for schema.");
+    TORCH_INTERNAL_ASSERT(
+        (rhs.index < getCorrectList(rhs.type).size()),
+        "Invalid index for schema.");
+
+    const Argument lhsArg = getCorrectList(lhs.type)[lhs.index];
+    const Argument rhsArg = getCorrectList(rhs.type)[rhs.index];
+    if (lhsArg.alias_info() && rhsArg.alias_info()) {
+      for (const auto& lhsSet : lhsArg.alias_info()->afterSets()) {
+        for (const auto& rhsSet : rhsArg.alias_info()->afterSets()) {
+          if (lhsSet == rhsSet) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  }
+  bool areAliasing(const SchemaArgument& lhs, const SchemaArgument& rhs, bool check_additional) const {
+    bool basic_check = areAliasing(lhs, rhs);
+    if (check_additional) {
+      const c10::Argument lhsArg = getCorrectList(lhs.type)[lhs.index];
+      const c10::Argument rhsArg = getCorrectList(rhs.type)[rhs.index];
+      if (lhsArg.alias_info() && lhsArg.alias_info()->isWildcardAfter()) {
+        c10::optional<std::vector<TypePtr>> lhsTypes = mapTypeToAliasTypeSet(lhsArg.type());
+        c10::optional<std::vector<TypePtr>> rhsTypes = mapTypeToAliasTypeSet(lhsArg.type());
+        if (lhsTypes && rhsTypes) {
+          std::unordered_set<TypePtr> flattened_types;
```

Comment:
is this the correct logic? I'm imagining a schema like:

```
fn(Tensor x(a -> *)) -> (Tensor(b -> *), Tensor)
```

In this case I think x and output 0 should alias, but x and output 1 should not alias

Reply:
From what I understand, x should alias both output 0 and output 1. This is because as noted in [alias_analysis.h](https://www.internalfb.com/code/fbsource/fbcode/caffe2/torch/csrc/jit/ir/alias_analysis.h) "we consider the wildcard set as potentially aliasing any value within the same type class" and both output 0 and 1 are in the tensor type class.

Reply:
I think that wildcard  set items can only alias other wildcard items. Did we ever get a confirmation from Elias about if this is the case?

Reply:
@Krovatkin has confirmed that the second output should not alias